src/services/completions.ts

```diff
@@ -2616,7 +2616,8 @@ namespace ts.Completions {
 
             // Need to insert 'this.' before properties of `this` type, so only do that if `includeInsertTextCompletions`
             if (preferences.includeCompletionsWithInsertText && scopeNode.kind !== SyntaxKind.SourceFile) {
-                const thisType = typeChecker.tryGetThisTypeAt(scopeNode, /*includeGlobalThis*/ false);
+                const thisType = typeChecker.tryGetThisTypeAt(scopeNode, /*includeGlobalThis*/ false,
+                    isClassLike(scopeNode.parent) ? scopeNode : getThisContainer(scopeNode, /*includeArrowFunctions*/ false));

```

Comment:
I would consider defaulting to `undefined` here rather than duplicating the code; the right side is what the type checker does by default if you don't provide the argument, so I think piggybacking on that is going to be the right choice.